text/0000-embroider-v2-package-format.md

```diff
@@ -0,0 +1,817 @@
+
+```js
+// This example:
+import { getOwnConfig, each, importSync } from "@ember/macros";
+let plugins = [];
+for (let plugin of each(getOwnConfig().registeredPlugins)) {
+  plugins.push(importSync(plugin).default);
+}
+
+// could compile to this, given OwnConfig
+// containing { registeredPlugins: ['@bigco/bar-chart', '@bigco/line-chart']}
+
+let plugins = [];
+plugins.push(importSync("@bigco/bar-chart").default);
+plugins.push(importSync("@bigco/line-chart").default);
+```
+
+It is a static error if the argument to `each` does not evaluate to a statically known array.
+
+### JavaScript Macro: moduleExists
+
+Allow you to test if an `import` (or `import()` or `importSync()`, since they all accept an argument with identical semantics) would succeed. Always compiles to a boolean literal.
+
+```js
+import { moduleExists, macroCondition, importSync } from "@ember/macros";
+if (macroCondition(moduleExists("ember-data"))) {
+  const DS = importSync("ember-data").default;
+  DS.Adapter.extend({
+    //
+  });
+}
+```
+
+Remember that you're always only allowed to `import` from your own **allowed dependencies**. So if an addon wants to optionally use another package only if that package is present, that package must be listed as an **Optional Peer Dependency**.
+
+`moduleExists` differs in one respect from all the other macros: there is no reliable build-time polyfill for it in non-Embroider builds. So when you use it in a non-Embroider build, you get a runtime implementation instead (equivalent to today's `require.has()`. If you need build-time branch elimination, try to use `dependencySatisfies` instead, which works correctly in both Embroider and non-Embroider builds.
+
+### JavaScript Macro: dependencySatisfies
+
+Allows you to test if the given **allowed dependency** satisfies the given semver range. Always compiles to a boolean literal.
+
+```js
+import { dependencySatisfies, macroCondition } from "@ember/macros";
+if (macroCondition(dependencySatisfies("ember-data", "^3.0.0"))) {
+  // include code here for ember-data 3.0 compat
+}
+```
+
+The package version will be `semver.coerce`d first, such that non-standard versions like "3.9.0-beta.0" will appropriately satisfy constraints like "> 3.8".
+
+### Javascript Macro: failBuild
+
+Allow you to cause a build failure with a custom error message. If `failBuild` isn't eliminated by `macroCondition`'s branch elimination, the build will fail.
+
+```js
+import { dependencySatisfies, macroCondition, failBuild } from "@ember/macros";
+if (macroCondition(dependencySatisfies("ember-data", "^3.0.0"))) {
+  // include code here for ember-data 3.0 compat
+} else {
+  failBuild(`We don't support ember-data versions other than ^3.0.0`);
+}
+```
+
+### Handlebars Macro: macroGetOwnConfig
```

Comment:
The Handlebars macro `macroGetConfig` is not mentioned in the RFC, I assume by mistake!?